src/perform_step/bdf_perform_step.jl

```diff
@@ -454,3 +454,129 @@ function perform_step!(integrator,cache::QNDF2Cache,repeat_step=false)
   cache.newton_iters = iter
   f(integrator.fsallast, u, p, t+dt)
 end
+
+function initialize!(integrator, cache::QNDFConstantCache)
+  integrator.kshortsize = 2
+  integrator.k = typeof(integrator.k)(integrator.kshortsize)
+  integrator.fsalfirst = integrator.f(integrator.uprev, integrator.p, integrator.t) # Pre-start fsal
+
+  # Avoid undefined entries if k is an array of arrays
+  integrator.fsallast = zero(integrator.fsalfirst)
+  integrator.k[1] = integrator.fsalfirst
+  integrator.k[2] = integrator.fsallast
+end
+
+function perform_step!(integrator,cache::QNDFConstantCache,repeat_step=false)
+  @unpack t,dt,uprev,u,f,p = integrator
+  @unpack udiff,dts,k,max_order,D,D2,R,U = cache
+  cnt = integrator.iter
+  κ = integrator.alg.kappa[k]
+  γ = inv((1-κ)*γₖ[k])
+  flag = true
+  for i in 2:k
+    if dts[i] != dts[1]
+      flag = false
+      break
+    end
+  end
+  if cnt > 2
+    if flag
+      ρ = dt/dts[1]
+      # backward diff
+      for i = 1:k
+        D2[1,i] = udiff[i]
+      end
+      backward_diff(D,D2,k)
+      if ρ != 1
+        U!(k,U)
+        R!(k,ρ,cache)
+        D .= D * (R * U)
+      end
+    else
+      for i = 1:k
+        D2[1,i] = udiff[i] * dt/dts[i]
+      end
+      backward_diff(D,D2,k)
+    end
+  else
+    γ = 1//1
+  end
+  # precalculations
+  u₀ = uprev + sum(D)
+  ϕ = zero(γ)
+  for i = 1:k
+    ϕ += γₖ[i]*D[i]
+  end
+  ϕ *= γ
+  tmp = u₀ - ϕ
+  γdt = γ*dt
+  W = calc_W!(integrator, cache, γdt, repeat_step)
+  # initial guess
+  z = dt*integrator.fsalfirst
+
+  z, η, iter, fail_convergence = diffeq_nlsolve!(integrator, cache, W, z, tmp, γ, 1, Val{:newton})
+  fail_convergence && return
+  u = tmp + γ*z
+
+  if integrator.opts.adaptive
+    if integrator.success_iter == 0
+      integrator.EEst = one(integrator.EEst)
+    elseif integrator.success_iter == 1
+      utilde = (u - uprev) - (udiff[1] * dt/dts[1])
+      atmp = calculate_residuals(utilde, uprev, u, integrator.opts.abstol, integrator.opts.reltol, integrator.opts.internalnorm)
+      integrator.EEst = integrator.opts.internalnorm(atmp)
+    else
+      δ = u - uprev
+      for i = 1:k
+        δ -= D[i]
+      end
+      utilde = (κ*γₖ[k] + inv(k+1)) * δ
```

Comment:
This one can also be `tilde = u - u0` ?

Reply:
there is a factor of `(κ*γₖ[k] + inv(k+1))`
`u - u_0 = del^(k+1)y_n+k` but `utilde = C * del^(k+1)y_n+k` and `c` is  `(κ*γₖ[k] + inv(k+1))`.